File: frontend/src/app/models/todo-task.ts
export interface TodoTask {
  id: string;
  title: string;
  description: string;
  startTime: string;
  endTime: string;
  isCompleted: boolean;
  createdAt: string;
}

export interface CreateTaskPayload {
  title: string;
  description: string;
  startTime: string;
  endTime: string;
}

export interface UpdateTaskPayload extends CreateTaskPayload {
  isCompleted: boolean;
}

export type TaskStatus = 'Pending' | 'Completed' | 'Overdue';

export function computeStatus(task: TodoTask, now: Date = new Date()): TaskStatus {
  if (task.isCompleted) {
    return 'Completed';
  }
  return new Date(task.endTime).getTime() < now.getTime() ? 'Overdue' : 'Pending';
}

File: backend/Models/TodoTask.cs
namespace TodoApi.Models;

public class TodoTask
{
    public Guid Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public DateTime StartTime { get; set; }
    public DateTime EndTime { get; set; }
    public bool IsCompleted { get; set; }
    public DateTime CreatedAt { get; set; }
}
